diff --git a/wordsactivity.py b/wordsactivity.py
index ecedd61..6bcc408 100644
--- a/wordsactivity.py
+++ b/wordsactivity.py
@@ -53,6 +53,8 @@ import dictdmodel
 from roundbox import RoundBox
 from sugar3.speech import SpeechManager
 
+from sugarapp.widgets import SugarCompatibleActivity
+
 EMPTY_HTML = '<body bgcolor="#E5E5E5"></body>'
 _AUTOSEARCH_TIMEOUT = 1000
 _ESPEAK_TO_NEW_LANG_CODE = {
@@ -235,13 +237,13 @@ class FilterToolItem(Gtk.ToolButton):
         self.emit('changed', key)
 
 
-class WordsActivity(activity.Activity):
+class WordsActivity(SugarCompatibleActivity):
 
     """Words Activity as specified in activity.info"""
 
     def __init__(self, handle):
         """Set up the Words activity."""
-        activity.Activity.__init__(self, handle)
+        SugarCompatibleActivity.__init__(self, handle)
 
         self._dictd_data_dir = './dictd/'
         self._dictionaries = dictdmodel.Dictionaries(self._dictd_data_dir)
